base_view_inheritance_extension/models/ir_ui_view.py

```diff
@@ -181,13 +181,16 @@ def inheritance_handler_attributes_domain_add(self, source, specs):
                 else:
                     new_value = str(expression.AND([old_domain, new_domain]))
                 new_value = self.str2var_domain_text(new_value)
+                old_value = self.str2var_domain_text(str(old_domain))
```

Comment:
You are doing this 2 times. Check line 173

Reply:
Not really: i first need to parse the string that I've got in domain (line 173), to get rid of all extra formatting (spaces, tabs, line breaks), then I parse it again in a clean string (line 184) and use that to create the proper `new_value` (line 193).

Let me know if I've been clear enough, thanks!

Reply:
To give you more context (https://github.com/OCA/sale-workflow/actions/runs/9097954749/job/25007038324?pr=3135#step:8:736)
![immagine](https://github.com/OCA/server-tools/assets/42804718/4ec08e79-6c4d-420a-9525-cd193e6eec8b)

Here after the "and" there is the `new_value` which is the parsed old value with the parsed new value, which gives no problem at all.

But after the "or" there is an empty string which is just the `old_value` (not parsed)

With this PR we also parse the `old_value` to make sure that is properly formatted.

Reply:
Yes, but don't call two times the same method. Optimize the code.

Reply:
Hi @pedrobaeza, could you check now please?

Only removing line breaks seems to work fine